components/tikv_util/src/sys/mod.rs

```diff
@@ -1,5 +1,89 @@
 // Copyright 2017 TiKV Project Authors. Licensed under Apache-2.0.
 
+#[cfg(target_os = "linux")]
+mod cgroup;
+
+#[cfg(target_os = "linux")]
+pub mod sys_quota {
+    use super::super::config::KB;
+    use super::cgroup::CGroupSys;
+
+    pub struct SysQuota {
+        cgroup: CGroupSys,
+    }
+
+    impl SysQuota {
+        pub fn new() -> Self {
+            Self {
+                cgroup: CGroupSys::default(),
+            }
+        }
+
+        pub fn cpu_cores_quota(&self) -> usize {
+            let cpu_num = sysinfo::get_logical_cores();
+            let cgroup_quota = self.cgroup.cpu_cores_quota();
+            if cgroup_quota < 0 {
+                cpu_num
+            } else {
+                std::cmp::min(cpu_num, cgroup_quota as usize)
+            }
+        }
+
+        pub fn memory_limit_in_bytes(&self) -> u64 {
+            use sysinfo::SystemExt;
+            let mut system = sysinfo::System::new();
+            system.refresh_all();
+            let total_mem = system.get_total_memory() * KB;
+            let cgroup_memory_limits = self.cgroup.memory_limit_in_bytes();
+            if cgroup_memory_limits < 0 {
```

Comment:
```suggestion
            if cgroup_memory_limits <= 0 {
```
It's better to use `total_mem` if the `cgroup_memory_limits==0` because of we cannot make sure the the function `self.cgroup.memory_limit_in_bytes` always return a non-zero number.